be/src/exprs/vectorized/array_functions.cpp

```diff
@@ -318,23 +321,18 @@ class ArrayRemoveImpl {
         }
     }
 
-    static ColumnPtr _array_remove_generic(const Column& array, const Column& target) {
-        if (array.only_null()) {
-            auto result = ArrayColumn::create(array.clone_empty(), UInt32Column::create());
-            result->append_nulls(array.size());
-            return result;
+    static ColumnPtr _array_remove_generic(const ColumnPtr& array, const ColumnPtr& target) {
+        if (array->only_null()) {
+            return array;
         }
-        if (auto nullable = dynamic_cast<const NullableColumn*>(&array); nullable != nullptr) {
-            auto array_col = down_cast<const ArrayColumn*>(nullable->data_column().get());
-            auto result = _array_remove_non_nullable(*array_col, target);
+        if (auto nullable = std::dynamic_pointer_cast<NullableColumn>(array); nullable != nullptr) {
+            auto result =
+                    _array_remove_non_nullable(std::dynamic_pointer_cast<ArrayColumn>(nullable->data_column()), target);
```

Comment:
The performance of `down_cast` (`std::static_cast`) is much better than `std::dynamic_pointer_cast`. 
So maybe `down_cast` is better?

Reply:
fixed.